syntax/NeogitStatus.vim

```diff
@@ -3,8 +3,9 @@ if exists("b:current_syntax")
 endif
 
 syn match NeogitObjectId /^[a-z0-9]\{7} /
-syn match NeogitBranch /.*/ contained
-syn match NeogitRemote /[a-zA-Z/]/ contained
+syn match NeogitCommitMessage /.*/ contained
+syn match NeogitBranch /\S\+/ contained nextgroup=NeogitCommitMessage
+syn match NeogitRemote /\S\+/ contained nextgroup=NeogitCommitMessage
```

Comment:
@TimUntersberger Ok, decided to have another go and made it work with just syn groups. Thanks for pulling me back to that :D